src/mds/FSMapUser.h

```diff
@@ -0,0 +1,81 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2006 Sage Weil <sage@newdream.net>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ */
+
+#ifndef CEPH_FSMAPCOMPACT_H
+#define CEPH_FSMAPCOMPACT_H
+
+#include "mds/mdstypes.h"
+#include <map>
+#include <string>
+
+class FSMapUser {
+public:
+  struct fs_info_t {
+    fs_cluster_id_t cid;
+    std::string name;
+    fs_info_t() : cid(FS_CLUSTER_ID_NONE) {}
+    void encode(bufferlist& bl, uint64_t features) const;
+    void decode(bufferlist::iterator &bl);
+  };
+
+  epoch_t epoch;
+  fs_cluster_id_t legacy_client_fscid;
+  bool enable_multiple;
```

Comment:
What is the use case for a client to know about enable_multiple?  We may not need to include this field